_guides/plugin-creation.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: documentation
+title: Create a custom plugin
+description: Customize The Lounge using built-in plugin support, and share plugins with others
+---
+
+Plugins for The Lounge are npm packages hosted on [the npm registry](https://www.npmjs.com).
+The work similarly to themes. A theme is a plugin but not every plugin is a theme.
+
+In a directory named after your new plugin (for example, `thelounge-plugin-foo`), start by creating a new package with the following command:
+
+```
+yarn init -y
+```
+
+This will create a `package.json` file that you must edit as such:
+
+- Make sure `"name"` is the name of your package, `"thelounge-plugin-foo"`
+- Change the value of `"main"` to be `"package.json"`
+- Add a `"keywords"` to make sure your plugin is discoverable:
+  ```json
+  "keywords": [
+    "thelounge",
+    "thelounge-plugin"
+  ]
+  ```
+- Add a `"thelounge"` section with required metadata. `"name"` is the display name that will appear in the client settings:
+  ```json
+  "thelounge": {
+    "name": "Plugin Name",
+    "type": "plugin"
+  },
+  ```
+- Reference your main javascript file:
+  ```json
+  "main": "index.js",
+  ```
+- Add a dependency on `"thelounge"`
+  ```json
+  "dependencies": {
+    "thelounge": "3.0.1"
+  }
+  ```
+Although it is not required, we strongly recommend you also fill in the `"homepage"`, `"repository"`, and `"bugs"` sections.
+
+You can then place all your plugin code in your `index.js`.
+
+The Lounge defines an entry point for plugins, you can use it like this:
+```js
+module.exports = {
+    onServerStart: api => {
+        // do stuff with api
+    },
+};
+```
+Refer to [the API reference](/docs/api) to learn what you can do or start implementing your own custom command with 
+[the custom command guide](/guides/command-creation).
+
+## Helpful things for plugin developers

```

Comment:
I'm thinking these can be sorted by making some helper functions in the api, and then just having them in the API docs. We won't want to explain everything here, you know?

Reply:
A helper for sending a message would be cool, I require could be helpful for other stuff tho I guess? For ppl who no experience with node like me, that might help.

Reply:
Well, we don't want to make this seem "official" because then we have to support it, you know? The helper functions are about being able to abstract out the internal code, and if people decide to use internal stuff, and we change it, we don't have to worry about that because it's not supported. Putting it in a guide on our official docs makes it look official.